### ISSUE TITLE
The possibility of cloning entities

### ISSUE THREAD

**Issue by baikinkirill** [mikro-orm/mikro-orm#5820]
**Is your feature request related to a problem? Please describe.**
In our project, various user entities have revisions. These revisions contain a snapshot of the entity at a specific point in time. For example, when a field of an entity is edited, a new revision is created with all the old and new fields.

**Describe the solution you'd like**
It would be very convenient to have a `.clone` method, which would construct a similar SQL query:

```sql
INSERT INTO auto (brand, model)
SELECT auto.brand, 'awesome model'
FROM auto
WHERE auto.id = 4
```

This method will significantly speed up the creation of revisions, as less data will be transferred over the network.

**Describe alternatives you've considered**
As an alternative, I want to construct such a query myself. However, to do this, I need to map the entity fields to the database columns. Is there a method in EntityManager that can help me with this?

**Comment by B4nan** [comment]
Right now I would be more in favor of a QB support for this, which would be the first step anyway if we want to have something like `em.clone`.

What API would you expect behind that example?

**Comment by B4nan** [comment]
The `replicate` method from laravel seems to be doing something else, its for cloning an existing model you have in memory, this issue is about firing a query and handling this on database level instead.

**Comment by rubiin** [comment]
Yeah . Its just clonining a non persisted entity. I was implying it as an example

**Comment by B4nan** [comment]
Merged to v7.1 via #7365, dev version should be available shortly.
